src/core/Input/InputField/InputField.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import InputLabel from '@material-ui/core/InputLabel';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import FormControl from '@material-ui/core/FormControl';
+import { BaseInputProps } from '../base';
+import { useStyles } from './styles';
+
+type Variant = 'primary' | 'error' | 'success' | undefined;
+
+interface CustomInputProps extends BaseInputProps {
+  variant?: Variant;
+  label: string;
+  error: boolean;
```

Comment:
These are already present in the [Text Field API](https://material-ui.com/api/text-field/) which you have extended as `BaseInputProps`, thus no need to add types for these here.